### ISSUE TITLE
@TestCase decorator

### ISSUE THREAD

**Issue by Aboisier** [Testy/TestyTs#25]
The current test case syntax is pretty ugly in my opinion.

```ts
@Test('My test', [
      new TestCase('Two plus two is four', 2, 2, 4),
      new TestCase(`Minus one that's three`, 4, -1, 3)
])
quickMaths(a: number, b: number, result: number) {
    expect.toBeEqual(a + b, result);
}
``` 

I think we can improve this. Here are a few examples of how we could go about it:

I like this syntax, but I'm not sure where we would put the timeout parameter, and we lose the opportunity to have a custom root-test name.
```ts
@TestCase('Two plus two is four', 2, 2, 4)
@TestCase(`Minus one that's three`, 4, -1, 3)
quickMaths(a: number, b: number, result: number) {
    expect.toBeEqual(a + b, result);
}
``` 

This option solves the name and timeout parameters problem and is still more readable than the current implementation.

```ts
@Test()
@TestCase('Two plus two is four', 2, 2, 4)
@TestCase(`Minus one that's three`, 4, -1, 3)
quickMaths(a: number, b: number, result: number) {
    expect.toBeEqual(a + b, result);
}
```

**Comment by bryanmenard** [comment]
I like the proposed syntax. However, I think having a separate decorator for the timeout (`@Timeout`) would be a good solution, too.

I think it's cleaner for regular tests (avoids having to pass empty arguments when we need a timeout but no test name) and also solves this issue.

**Comment by Aboisier** [comment]
I really like this solution! I was definitely pretty annoyed by the empty argument.

**Comment by Aboisier** [comment]
Implemented in #29 .
